lib/observability/metrics/dynamo/api.go

```diff
@@ -0,0 +1,163 @@
+}
+
+func (m *APIMetrics) GetItemWithContext(ctx context.Context, input *dynamodb.GetItemInput, opts ...request.Option) (*dynamodb.GetItemOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.GetItemWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "get_item", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) PutItemWithContext(ctx context.Context, input *dynamodb.PutItemInput, opts ...request.Option) (*dynamodb.PutItemOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.PutItemWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "put_item", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) UpdateItemWithContext(ctx context.Context, input *dynamodb.UpdateItemInput, opts ...request.Option) (*dynamodb.UpdateItemOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.UpdateItemWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "update_item", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) DeleteTableWithContext(ctx context.Context, input *dynamodb.DeleteTableInput, opts ...request.Option) (*dynamodb.DeleteTableOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.DeleteTableWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "delete_table", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) BatchWriteItemWithContext(ctx context.Context, input *dynamodb.BatchWriteItemInput, opts ...request.Option) (*dynamodb.BatchWriteItemOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.BatchWriteItemWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "batch_write_item", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) ScanWithContext(ctx context.Context, input *dynamodb.ScanInput, opts ...request.Option) (*dynamodb.ScanOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.ScanWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "scan", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) CreateTableWithContext(ctx context.Context, input *dynamodb.CreateTableInput, opts ...request.Option) (*dynamodb.CreateTableOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.CreateTableWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "create_table", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) DescribeTableWithContext(ctx context.Context, input *dynamodb.DescribeTableInput, opts ...request.Option) (*dynamodb.DescribeTableOutput, error) {
+	start := time.Now()
+	output, err := m.DynamoDBAPI.DescribeTableWithContext(ctx, input, opts...)
+
+	recordMetrics(m.tableType, "create_table", err, time.Since(start).Seconds())
+
+	return output, err
+}
+
+func (m *APIMetrics) Query(input *dynamodb.QueryInput) (*dynamodb.QueryOutput, error) {
```

Comment:
Rather than adding a `Query` and `QueryWithContext`, can you update the few calls to `Query` to `QueryWithContext` instead, using a `context.TODO()` if needed? That way it'll be more consistent and we won't accidentally miss metrics in the future.